website/static/js/pointers.js

```diff
@@ -19,132 +19,238 @@ var SEARCH_MY_PROJECTS_SUBMIT_TEXT = 'Search my projects';
+    },
+    doneSearching: function(){
+      var self = this;
+      self.searchAllProjectsSubmitText(SEARCH_ALL_SUBMIT_TEXT);
+      self.searchMyProjectsSubmitText(SEARCH_MY_PROJECTS_SUBMIT_TEXT);
+      self.loadingResults(false);
     },
     searchAllProjects: function() {
-        this.includePublic(true);
-        this.pageToGet(0);
-        this.searchAllProjectsSubmitText('Searching...');
-        this.fetchResults();
+        var self = this;
+        self.includePublic(true);
+        self.pageToGet(0);
+        self.searchAllProjectsSubmitText('Searching...');
+        self.loadingResults(true);
+        self.fetchResults();
     },
     searchMyProjects: function() {
-        this.includePublic(false);
-        this.pageToGet(0);
-        this.searchMyProjectsSubmitText('Searching...');
-        this.fetchResults();
+        var self = this;
+        self.includePublic(false);
+        self.pageToGet(0);
+        self.searchMyProjectsSubmitText('Searching...');
+        self.loadingResults(true);
+        self.fetchResults();
     },
     fetchResults: function() {
         var self = this;
         self.errorMsg('');
         self.searchWarningMsg('');
-
-        if (self.query()) {
-            self.results([]); // clears page for spinner
-            self.loadingResults(true); // enables spinner
-
-            osfHelpers.postJSON(
-                '/api/v1/search/node/', {
-                    query: self.query(),
-                    nodeId: nodeId,
-                    includePublic: self.includePublic(),
-                    page: self.pageToGet()
-                }
-            ).done(function(result) {
-                if (!result.nodes.length) {
-                    self.errorMsg('No results found.');
-                } else {
-                    result.nodes.forEach(function(each) {
-                        if (each.isRegistration) {
-                            each.dateRegistered = new osfHelpers.FormattableDate(each.dateRegistered);
+        self.results([]); // clears page for spinner
+        self.selection([]);
+        var query = '', myProjects = '', pageNum = self.pageToGet()+1;
+        if (self.query()){
+            query += 'filter[title]='+self.query()+'&';
+        }
+        if (!self.includePublic()){
+            myProjects = 'users/me/';
+        }
+        var url = osfHelpers.apiV2Url(myProjects+self.inputType()+'/', {query: query+'page='+pageNum+'&embed=contributors&page[size]=4'});
+        var request = osfHelpers.ajaxJSON(
+            'GET',
+            url,
+            {'isCors': true});
+        request.done(function(response) {
+            var nodes = response.data;
+            if (!nodes.length) {
+                self.errorMsg('No results found.');
+            }
+            else {
+                var count = nodes.length;
+                var url = osfHelpers.apiV2Url('nodes/'+nodeId+'/node_links/', {});
+                var request = osfHelpers.ajaxJSON(
```

Comment:
This should probably renamed since it is used multiple places, shouldn't be a real problem just for readability